Frontend/client.py

```diff
@@ -103,5 +97,53 @@ def status(uid: str) -> Status:
         raise Exception('Server is down.')
 
 
-# test upload:
-# print(upload(os.path.join(upload_folder, 'asyncio-intro.pptx')))
+def status_by_email_and_filename(email: str, filename: str) -> Status:
+    """
+    @summary:
+        Sends the appropriate HTTP request to the web app with the email and filename in the request body.
+        and returns a Status object with the data from the response.
+    @param email:
+        The email of the user.
+    @param filename:
+        The name of the file.
+    @return:
+        Status object with the data from the response.
+    @raise:
+        Exception: if the server is down.
+        Exception: if the server returned an error.
+    """
+    # Prepare the request URL
+    url = 'http://localhost:5000/content/json/'
+
+    # Prepare the data to send in the body
+    form_data = {
+        'email': email,
+        'filename': filename
+    }
+
+    try:
+        # Send the GET request with the data in the body
+        response = requests.get(url, data=form_data)
+
+        # case request was successful:
+        if response.status_code == 200 or response.status_code == 202:
+            response_json = response.json()
+            response_status = response_json.get('status', '')
+            filename = response_json.get('filename', '')
+            timestamp = response_json.get('upload time', '')
+            explanation = response_json.get('explanation', '')
+            return Status(response_status, filename, timestamp, explanation)
+
+        # case 404 or 400 error:
+        elif response.status_code == 404 or response.status_code == 400:
+            response_json = response.json()
+            response_status = response_json.get('status', '')
+            return Status(response_status)
+
+        # case an error occurred:
+        elif 400 <= response.status_code < 500:
+            # server error message: "error"
+            raise Exception('Server error.')
+
+    except requests.exceptions.ConnectionError:
+        raise Exception('Server is down.')
```

Comment:
Always throw the most specific exception -
https://stackoverflow.com/questions/2052390/manually-raising-throwing-an-exception-in-python
https://docs.python.org/3/library/exceptions.html